entities/storobj/storage_object.go

```diff
@@ -206,15 +217,20 @@ func ObjectsByDocID(bucket bucket, ids []uint64,
 		docIDBuf = make([]byte, 8)
 		out      = make([]*Object, len(ids))
 		i        = 0
+		lsmBuf   = bufPool.Get()
 	)
 
+	defer bufPool.Put(lsmBuf)
```

Comment:
does it really return new buffer? `lsmBuf` evaluated here would point to old buffer.

Reply:
btw: if it is possible to return different object than the one taken from pool, maybe both of them could be returned for future usage, since memory is already allocated twice.

Reply:
> btw: if it is possible to return different object than the one taken from pool, maybe both of them could be returned for future usage, since memory is already allocated twice.

I think I prefer only returning the larger buffer, because that way all buffers in the pool grow over time if needed. If we keep both sizes, but the request pattern always requires a larger buffer, half of the buffers in the pool would be unusable. So you'd only have a 50% chance of getting one that you can use. But if we only return the grown buffer, at some point all smaller buffers should be replaced and you will be guaranteed to only get larger buffers after a while.

Reply:
that makes sense, thx for the explanation.